hubblestack/extmods/grains/cloud_details.py

```diff
@@ -12,15 +14,16 @@ def get_cloud_details():
     grains = {}
 
     aws = _get_aws_details()
-    azure = _get_azure_details()
-    gcp = _get_gcp_details()
-
-    if aws['cloud_details']:
+    if not aws['cloud_details']: # Unable to fetch details from AWS. Let's try from Azure
+        azure = _get_azure_details()
```

Comment:
you can add debug log statements here for future debugging cases. Similarly, you can add debug statements at multiple points in the code below as well.

Reply:
added debug statements